pyproject.toml

```diff
@@ -12,39 +13,93 @@ classifiers = [
     "Programming Language :: Python :: 3.12",
     "Programming Language :: Python :: 3.11",
 ]
+requires-python = ">=3.11"
+dependencies = [
+    "venusian~=3.1.0"
+]
 
-[tool.poetry.urls]
+[tool.hatch.urls]
 Changelog = "https://github.com/pauleveritt/hopscotch/releases"
 
-[tool.poetry.dependencies]
-python = "^3.11"
-venusian = "^3.1.0"
-
-[tool.poetry.dev-dependencies]
-pytest = "^7.4.3"
-nox = "^2023.4.22"
-coverage = {extras = ["toml"], version = "^7.3", allow-prereleases = true}
-safety = "^2.3.5"
-mypy = "^1.7"
-xdoctest = {extras = ["colors"], version = "^1.1.2"}
-sphinx = "^4.4.0"
-sphinx-autobuild = "^2021.3.14"
-pre-commit = "^3.5.0"
-flake8 = "^6.1.0"
-black = "^22.1"
-flake8-bandit = "^4.1.1"
-flake8-bugbear = "^23.9.16"
-flake8-docstrings = "^1.6.0"
-flake8-rst-docstrings = "^0.3.0"
-pep8-naming = "^0.13.3"
-darglint = "^1.8.1"
-reorder-python-imports = "^3.12.0"
-pre-commit-hooks = "^4.0.1"
-Pygments = "^2.11.2"
-nox-poetry = "^1.0.3"
-furo = "^2022.3.4"
-myst-parser = "^0.17.0"
-sybil = "^3.0.1"
+[tool.hatch.envs.pre-commit]
+dependencies = [
+    "black",
+    "darglint",
+    "flake8",
+    "flake8-bandit",
+    "flake8-bugbear",
+    "flake8-docstrings",
+    "flake8-rst-docstrings",
+    "pep8-naming",
+    "pre-commit",
+    "pre-commit-hooks",
+    "reorder-python-imports"
+]
+[tool.hatch.envs.mypy]
+dependencies = [
+    "mypy",
+    "pytest"
+]
+[tool.hatch.envs.test]
+dependencies = [
+    "pytest~=7.4.3",
+    "nox~=2023.4.22",
+    "safety~=2.3.5",
+    "mypy~=1.7",
+    "sphinx~=4.4.0",
+    "sphinx-autobuild~=2021.3.14",
+    "pre-commit~=3.5.0",
+    "flake8~=6.1.0",
+    "black~=22.1",
+    "flake8-bandit~=4.1.1",
+    "flake8-bugbear~=23.9.16",
+    "flake8-docstrings~=1.6.0",
+    "flake8-rst-docstrings~=0.3.0",
+    "pep8-naming~=0.13.3",
+    "darglint~=1.8.1",
+    "reorder-python-imports~=3.12.0",
+    "pre-commit-hooks~=4.0.1",
+    "Pygments~=2.11.2",
+    "furo~=2022.3.4",
+    "myst-parser~=0.17.0",
+    "sybil~=3.0.1",
+    "coverage[toml]~=7.3",
+    "xdoctest[colors]~=1.1.2",
+    "pytest-cov~=4.1.0",
```

Comment:
Is there a reason you used `coverage` instead of `pytest-cov`? I added it because I use `pytest-cov` all the time.

Reply:
I inherited decisions from this [project. and cookie cutter](https://github.com/cjolowicz/cookiecutter-hypermodern-python). He has a long set of docs on all this and an upcoming O'Reilly book.

BUT...I now disagree with some of his decisions. So I'm 100% ok with the change you suggest.